src/journal/journald-native.c

```diff
@@ -63,7 +63,7 @@ static void server_process_entry_meta(
                  startswith(p, "SYSLOG_IDENTIFIER=")) {
                 char *t;
 
-                t = strndup(p + 18, l - 18);
+                t = memdup_suffix0(p + 18, l - 18);
```

Comment:
I should probably take a look at how t is used. If it's considered a string everywhere else we still lose the bytes after the first null byte in the middle of it. Maybe it should be escaped.

Reply:
Oh, sorry. It just copies "SYSLOG_IDENTIFIER=" :-)

Reply:
OK. I managed to send "MESSAGE":[104,101,121,0,104,111]," and "SYSLOG_IDENTIFIER":[72,69,89,0,72,79]" with `printf 'MESSAGE=hey\0ho\nSYSLOG_IDENTIFIER=HEY\0HO\n' | nc -w1 -u -U /run/systemd/journal/socket` so it seems to be working.